src/aggregator/aggregator/leader_flush_mgr.go

```diff
@@ -181,20 +176,39 @@ func (mgr *leaderFlushManager) Prepare(buckets []*flushBucket) (flushTask, time.
 		}
 	}
 
-	durationSinceLastPersist := time.Duration(nowNanos - mgr.lastPersistAtNanos)
```

Comment:
weird that we were updating during the Prepare because we haven't actually flushed. nothing is preventing the leader from crashing after this but before flushing.

Reply:
I thought so as well, but thankfully this didn't relate to any prepared flush task from this run, so it would've only updated with the results of any previous flushes, and any crashes would require the follower to re-flush rather than persisting the wrong flush times

It certainly took a few reads to understand, was fairly confusingly written, hopefully this approach is clearer